Terminal.Gui/View/View.Navigation.cs

```diff
@@ -619,7 +639,7 @@ public bool SetFocus ()
 
         if (Arrangement.HasFlag (ViewArrangement.Overlapped))
         {
-            SuperView?.MoveSubViewToEnd (this);
+            SuperView?.MoveOverlappedSubViewToEnd (this);
```

Comment:
I suggest you nuke MoveOverlappedSubViewToEnd and put this flag check inside MoveubViewToEnd.

Will that work?

Reply:
No they are different. `MoveSubViewToEnd` is intended to change the sub-views order and `MoveOverlappedSubViewToEnd` always maintain the same order. Otherwise you haven't a way to change the sub-views order.